src/imitation/util/logger.py

```diff
@@ -155,6 +160,46 @@ def close(self):
             logger.close()
 
 
+class WandbOutputFormat(sb_logger.KVWriter):
+    """A stable-baseline logger that writes to wandb.
+
+    Users need to call `wandb.init()` before initializing `WandbOutputFormat`.
+    """
+
+    def __init__(self):
+        """Initializes an instance of WandbOutputFormat.
+
+        Raises:
+            ModuleNotFoundError: wandb is not installed.
+        """
+        try:
+            import wandb
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "Trying to log data with `WandbOutputFormat` "
+                "but `wandb` not installed: try `pip install wandb`.",
+            )
+        self.wandb_module = wandb
+
+    def write(
+        self,
+        key_values: Dict[str, Any],
+        key_excluded: Dict[str, Union[str, Tuple[str, ...]]],
+        step: int = 0,
+    ) -> None:
+        for (key, value), (_, excluded) in zip(
```

Comment:
> I think this only works if key_values and key_excluded have the same keys. Is that always meant to be true? If so, you could add an assertion to check this.

You're right. I borrowed this from how SB3 implements [HumanOutputFormat](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/common/logger.html#HumanOutputFormat.write). The keys should be meant to be true because when [`Logger.record`](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/common/logger.html#Logger.record), it does the following to make sure the keys match each other exactly.

```python
self.name_to_value[key] = value
self.name_to_excluded[key] = exclude
```

Reply:
But it doesn't hurt to do an assertion check.